### PR TITLE
Fix wave restarts and GEFS FHOUT/FHMAX

### LINKED ISSUE #3001: Possible bug on GEFS fcst segment
### What is wrong?

When run gefs in segment, the fcst hour seems overlapped, as below:

[Wei.Huang@hfe03 2021032312]$ grep cfhour fcst_mem00*

1. fcst_mem000_seg0.log: 6:  in wrt run, nfhour=  0.333333333333333       cfhour=000
2. fcst_mem000_seg0.log: 6:  in wrt run, nfhour=   6.00000000000000       cfhour=006
3. fcst_mem000_seg0.log: 6:  in wrt run, nfhour=   12.0000000000000       cfhour=012
4. fcst_mem000_seg0.log: 6:  in wrt run, nfhour=   18.0000000000000       cfhour=018
5. fcst_mem000_seg0.log: 6:  in wrt run, nfhour=   24.0000000000000       cfhour=024
6. fcst_mem000_seg0.log: 6:  in wrt run, nfhour=   30.0000000000000       cfhour=030
7. fcst_mem000_seg0.log: 6:  in wrt run, nfhour=   36.0000000000000       cfhour=036
8. fcst_mem000_seg0.log: 6:  in wrt run, nfhour=   42.0000000000000       cfhour=042
9. fcst_mem000_seg0.log: 6:  in wrt run, nfhour=   48.0000000000000       cfhour=048

fcst_mem000_seg1.log: 6:  in wrt run, nfhour=   12.0000000000000       cfhour=012
fcst_mem000_seg1.log: 6:  in wrt run, nfhour=   24.0000000000000       cfhour=024
fcst_mem000_seg1.log: 6:  in wrt run, nfhour=   36.0000000000000       cfhour=036
fcst_mem000_seg1.log: 6:  in wrt run, nfhour=   48.0000000000000       cfhour=048
fcst_mem000_seg1.log: 6:  in wrt run, nfhour=   54.0000000000000       cfhour=054
fcst_mem000_seg1.log: 6:  in wrt run, nfhour=   60.0000000000000       cfhour=060
fcst_mem000_seg1.log: 6:  in wrt run, nfhour=   66.0000000000000       cfhour=066
fcst_mem000_seg1.log: 6:  in wrt run, nfhour=   72.0000000000000       cfhour=072
fcst_mem000_seg1.log: 6:  in wrt run, nfhour=   78.0000000000000       cfhour=078
fcst_mem000_seg1.log: 6:  in wrt run, nfhour=   84.0000000000000       cfhour=084
fcst_mem000_seg1.log: 6:  in wrt run, nfhour=   90.0000000000000       cfhour=090
fcst_mem000_seg1.log: 6:  in wrt run, nfhour=   96.0000000000000       cfhour=096
fcst_mem000_seg1.log: 6:  in wrt run, nfhour=   102.000000000000       cfhour=102
fcst_mem000_seg1.log: 6:  in wrt run, nfhour=   108.000000000000       cfhour=108
fcst_mem000_seg1.log: 6:  in wrt run, nfhour=   114.000000000000       cfhour=114
fcst_mem000_seg1.log: 6:  in wrt run, nfhour=   120.000000000000       cfhour=120

- fcst_mem001_seg0.log: 6:  in wrt run, nfhour=  0.333333333333333       cfhour=000
- fcst_mem001_seg0.log: 6:  in wrt run, nfhour=   6.00000000000000       cfhour=006
- fcst_mem001_seg0.log: 6:  in wrt run, nfhour=   12.0000000000000       cfhour=012
- fcst_mem001_seg0.log: 6:  in wrt run, nfhour=   18.0000000000000       cfhour=018
- fcst_mem001_seg0.log: 6:  in wrt run, nfhour=   24.0000000000000       cfhour=024
- fcst_mem001_seg0.log: 6:  in wrt run, nfhour=   30.0000000000000       cfhour=030
- fcst_mem001_seg0.log: 6:  in wrt run, nfhour=   36.0000000000000       cfhour=036
- fcst_mem001_seg0.log: 6:  in wrt run, nfhour=   42.0000000000000       cfhour=042
- fcst_mem001_seg0.log: 6:  in wrt run, nfhour=   48.0000000000000       cfhour=048
- fcst_mem001_seg0.log: 6:  in wrt run, nfhour=   54.0000000000000       cfhour=054
- fcst_mem001_seg0.log: 6:  in wrt run, nfhour=   60.0000000000000       cfhour=060
- fcst_mem001_seg0.log: 6:  in wrt run, nfhour=   66.0000000000000       cfhour=066
- fcst_mem001_seg0.log: 6:  in wrt run, nfhour=   72.0000000000000       cfhour=072
- fcst_mem001_seg0.log: 6:  in wrt run, nfhour=   78.0000000000000       cfhour=078
- fcst_mem001_seg0.log: 6:  in wrt run, nfhour=   84.0000000000000       cfhour=084
- fcst_mem001_seg0.log: 6:  in wrt run, nfhour=   90.0000000000000       cfhour=090
- fcst_mem001_seg0.log: 6:  in wrt run, nfhour=   96.0000000000000       cfhour=096
- fcst_mem001_seg0.log: 6:  in wrt run, nfhour=   102.000000000000       cfhour=102
- fcst_mem001_seg0.log: 6:  in wrt run, nfhour=   108.000000000000       cfhour=108
- fcst_mem001_seg0.log: 6:  in wrt run, nfhour=   114.000000000000       cfhour=114
- fcst_mem001_seg0.log: 6:  in wrt run, nfhour=   120.000000000000       cfhour=120

fcst_mem001_seg1.log: 6:  in wrt run, nfhour=   12.0000000000000       cfhour=012
fcst_mem001_seg1.log: 6:  in wrt run, nfhour=   24.0000000000000       cfhour=024
fcst_mem001_seg1.log: 6:  in wrt run, nfhour=   36.0000000000000       cfhour=036
fcst_mem001_seg1.log: 6:  in wrt run, nfhour=   48.0000000000000       cfhour=048
fcst_mem001_seg1.log: 6:  in wrt run, nfhour=   54.0000000000000       cfhour=054
fcst_mem001_seg1.log: 6:  in wrt run, nfhour=   60.0000000000000       cfhour=060
fcst_mem001_seg1.log: 6:  in wrt run, nfhour=   66.0000000000000       cfhour=066
fcst_mem001_seg1.log: 6:  in wrt run, nfhour=   72.0000000000000       cfhour=072
fcst_mem001_seg1.log: 6:  in wrt run, nfhour=   78.0000000000000       cfhour=078
fcst_mem001_seg1.log: 6:  in wrt run, nfhour=   84.0000000000000       cfhour=084
fcst_mem001_seg1.log: 6:  in wrt run, nfhour=   90.0000000000000       cfhour=090
fcst_mem001_seg1.log: 6:  in wrt run, nfhour=   96.0000000000000       cfhour=096
fcst_mem001_seg1.log: 6:  in wrt run, nfhour=   102.000000000000       cfhour=102
fcst_mem001_seg1.log: 6:  in wrt run, nfhour=   108.000000000000       cfhour=108
fcst_mem001_seg1.log: 6:  in wrt run, nfhour=   114.000000000000       cfhour=114
fcst_mem001_seg1.log: 6:  in wrt run, nfhour=   120.000000000000       cfhour=120

For mem000, seg 0 fcst from 00 - 48, then seg 1 from 12 to 120, should seg 1 be from 48 - 120?
For mem001 and mem002, seg 0 from 00 - 120, and then seg 1 from 12 to 120, seg 1 here is not needed at all, right?

rocotostat shows this:

+ /apps/rocoto/1.3.7/bin/rocotostat -d c48gefs.db -w c48gefs.xml
       CYCLE                    TASK                       JOBID               STATE         EXIT STATUS     TRIES      DURATION
================================================================================================================================
202103231200                stage_ic                      803100           SUCCEEDED                   0         1          21.0
202103231200               wave_init                      803099           SUCCEEDED                   0         1          28.0
202103231200          prep_emissions                      803098           SUCCEEDED                   0         1          17.0
202103231200        fcst_mem000_seg0                      803195           SUCCEEDED                   0         1        1164.0
202103231200        fcst_mem000_seg1                      803964           SUCCEEDED                   0         1        2812.0
202103231200        fcst_mem001_seg0                      803196           SUCCEEDED                   0         1        2859.0
202103231200        fcst_mem001_seg1                      805320           SUCCEEDED                   0         1        2890.0
202103231200        fcst_mem002_seg0                      803197           SUCCEEDED                   0         1        2850.0
202103231200        fcst_mem002_seg1                      805321           SUCCEEDED                   0         1        2884.0

it seems mem000 over-used 1/3 of CPU, and mem001, and mem002 doubled the CPU cost.

### What should have happened?

We expect all members, if in two seg fcst, it should be:
seg 0, fcst from 00 -> 48,
seg 1, fcst from 48 -> 120.

### What machines are impacted?

All or N/A

### What global-workflow hash are you using?

The test is using EPIC's fork of global-workflow, which is point to the current develop.

### Steps to reproduce

To produce on Hera:
1. compile as: build_all.sh -w
2. configure with:
    HPC_ACCOUNT=epic \
        pslot=c48gefs \
        RUNTESTS=/scratch1/NCEPDEV/stmp2/$USER/GEFSTESTS \
        ./workflow/create_experiment.py \
        --yaml ci/cases/pr/C48_S2SWA_gefs.yaml
3. start crontab.

### Additional information

COMROOT and EXPDIR on Hera at:

[Wei.Huang@hfe03 GEFSTESTS]$ pwd
/scratch1/NCEPDEV/stmp2/Wei.Huang/GEFSTESTS
[Wei.Huang@hfe03 GEFSTESTS]$ ls -l
total 8
drwxr-sr-x 3 Wei.Huang stmp 4096 Oct 10 22:56 COMROOT
drwxr-sr-x 3 Wei.Huang stmp 4096 Oct 10 22:56 EXPDIR
[Wei.Huang@hfe03 GEFSTESTS]$ ls -l *
COMROOT:
total 4
drwxr-sr-x 4 Wei.Huang stmp 4096 Oct 10 22:57 c48gefs

EXPDIR:
total 4
drwxr-sr-x 3 Wei.Huang stmp 4096 Oct 11 14:10 c48gefs

### Do you have a proposed solution?

No

### PR DESCRIPTION
**Pull request by WalterKolczynski-NOAA**
# Description
Fixes some issues that were preventing wave restarts from operating correctly.

First, the wave restart files were not being correctly linked from `$DATA` to `$DATArestart`. The files are placed in the root of `$DATA` instead of in `${DATA}/WAVE_RESTART`, so now links for the individual files are created.

Second, the incorrect filenames were being searches for and copied as part of a rerun. Filenames were geared towards multigrid waves, which use the grid names, but single grid just uses a `ww3`. Since multigrid waves are deprecated in workflow and will soon be removed (#2637), these were updated only supporting the single-grid option.

These fixes allow forecast segments (and emergency restarts) to work correctly when waves are on.

Additionally, the `FHMAX` (and `FHOUT`) for perturbed GEFS members was being overwritten by `config.efcs` due to remnant code from when it was copied from the EnKF version. This interfered with the segment setting for those GEFS members.

Resolves #3001

# Type of change
- [x] Bug fix (fixes something broken)
- [ ] New feature (adds functionality)
- [ ] Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO
- Does this change require an update to any of the following submodules? NO

# How has this been tested?
- S2SW forecast-only test with segments on Hercules

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have documented my code, including function, input, and output descriptions
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] This change is covered by an existing CI test or a new one has been added
- [x] I have made corresponding changes to the system documentation if necessary

### CODE REVIEW

**Comment by WalterKolczynski-NOAA** [comment]
ATTN: @weihuang-jedi

File: ush/forecast_postdet.sh [hunk 1]
```diff
@@ -340,7 +340,7 @@ WW3_postdet() {
     echo "Copying WW3 restarts for 'RUN=${RUN}' at '${restart_date}' from '${restart_dir}'"
     local ww3_restart_file
     for ww3_grid in ${waveGRD} ; do
-      ww3_restart_file="${restart_dir}/${restart_date:0:8}.${restart_date:8:2}0000.restart.${ww3_grid}"
+      ww3_restart_file="${restart_dir}/${restart_date:0:8}.${restart_date:8:2}0000.restart.ww3"
```
Comment (aerorahul) [inline ush/forecast_postdet.sh]: This line does not have a variables `ww3_grid`, so there is no need to loop

File: ush/forecast_postdet.sh [hunk 2]
```diff
@@ -361,6 +361,15 @@ WW3_postdet() {
...
   fi  # [[ "${warm_start}" == ".true." ]]
 
+  # Link restart files
+  local restart_file
+  # Use restart_date if it was determined above, otherwise use initialization date
+  for (( vdate = "${restart_date:-${PDY}${cyc}}";  vdate <= forecast_end_cycle;
+         vdate = $(date --utc -d "${vdate:0:8} ${vdate:8:2} + ${restart_interval} hours" +%Y%m%d%H) )); do
+    restart_file="${vdate:0:8}.${vdate:8:2}0000.restart.ww3"
+    ${NLN} "${DATArestart}/WW3_RESTART/${restart_file}" "${restart_file}"
+  done
```
Comment (aerorahul) [inline ush/forecast_postdet.sh]: I think we should split this into 2 parts.
part 1 should be linking the restart for initialization.  This is either the restarts at `restart_date` (if found above) or `PDYcyc`.
If it is `PDYcyc`, the `restart_dir` is `COMIN_WAVE_RESTART`.  If it is a `RERUN=YES`, it is `DATArestart/WW3_RESTART`.  This restart **must** exist.

part 2 is creating the links from `restart_date+restart_interval` to `forecast_end_cycle`
These are the restarts that will be created as the model integrates.

**Review by aerorahul — CHANGES_REQUESTED** [review]
I think there is an issue in the `forecast_postdet.sh` part.  Also a suggestion to break that into 2 parts.

File: ush/forecast_postdet.sh (hunk 2 above)
Comment (WalterKolczynski-NOAA) [reply ush/forecast_postdet.sh]: There's already a separate part to copy the initial restart in, if necessary. Just need to adjust the start date for the links when that happens.

**Comment by WalterKolczynski-NOAA** [comment]
Will need to add in additional fixes for GEFS perturbed members, as they are ignoring the segments.

**Review by weihuang-jedi — APPROVED** [review]

**Comment by weihuang-jedi** [comment]
I have test the PR code on Hercules. I worked as expected. Fixed the segments issue.
Thanks @WalterKolczynski-NOAA !

**Comment by TerrenceMcGuinness-NOAA** [comment]
Jenkins controller had technical difficulties and was off line.  Clearing Hercules is under currently under maintenance.

**Comment by WalterKolczynski-NOAA** [comment]
I need to make additional changes. The ICs and stage IC job is expecting wave restarts to use the wave grid, but when not using multigrid the files should actually be `ww3`.

**Comment by DavidHuber-NOAA** [comment]
> Experiment C96C48_hybatmaerosnowDA **FAILED** on Hera in Build# 1 with error logs:
> 
> ```
> /scratch1/NCEPDEV/global/CI/3009/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA_4c3ac105/logs/2021122018/gdas_aeroanlinit.log
> /scratch1/NCEPDEV/global/CI/3009/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA_4c3ac105/logs/2021122018/gfs_aeroanlinit.log
> ```
> 
> Follow link here to view the contents of the above file(s): [(link)](https://gist.github.com/emcbot/f18065fe4825d02be54c101c3f99d178)

I am receiving the same error on WCOSS2.  There is a mismatch between the `viirs_n20` name in the GDA: `gfs.tCCz.viirs_n20_aod.tm00.nc` and the naming convention in the [Jinja yaml in the GDASApp](https://github.com/NOAA-EMC/GDASApp/blob/c5ff4e7e3bd17532211d077259ba8c7e16e5ffbb/parm/aero/obs/config/viirs_n20_aod.yaml.j2#L6).

It looks like this will be fixed by https://github.com/NOAA-EMC/global-workflow/pull/2978.

**Comment by WalterKolczynski-NOAA** [comment]
> > Experiment C96C48_hybatmaerosnowDA **FAILED** on Hera in Build# 1 with error logs:
> > ```
> > /scratch1/NCEPDEV/global/CI/3009/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA_4c3ac105/logs/2021122018/gdas_aeroanlinit.log
> > /scratch1/NCEPDEV/global/CI/3009/RUNTESTS/COMROOT/C96C48_hybatmaerosnowDA_4c3ac105/logs/2021122018/gfs_aeroanlinit.log
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Follow link here to view the contents of the above file(s): [(link)](https://gist.github.com/emcbot/f18065fe4825d02be54c101c3f99d178)
> 
> I am receiving the same error on WCOSS2. There is a mismatch between the `viirs_n20` name in the GDA: `gfs.tCCz.viirs_n20_aod.tm00.nc` and the naming convention in the [Jinja yaml in the GDASApp](https://github.com/NOAA-EMC/GDASApp/blob/c5ff4e7e3bd17532211d077259ba8c7e16e5ffbb/parm/aero/obs/config/viirs_n20_aod.yaml.j2#L6).
> 
> It looks like this will be fixed by #2978.

This is actually an issue with the IC names for wave restarts. I made symlinks on Hera that need to by sync'd over.

**Review by DavidHuber-NOAA — DISMISSED** [review]

**Comment by DavidHuber-NOAA** [comment]
I'm going to manually run just the GEFS tests on Hera.

**Comment by DavidHuber-NOAA** [comment]
All tests passed on Hera, but after a conversation with Walter and Kate, we will wait to merge this until an additional fix for the stage_ic job goes in.

**Review by DavidHuber-NOAA — APPROVED** [review]

**Pull request merged**